### PR TITLE
Add Marginalia migration guide for Rails 7 upgrade

### PR DESCRIPTION
**Pull request by keeran**
Adds a guide for migrating to native `ActiveRecord::QueryLogs` from Marginalia when upgrading to Rails 7.

/cc @kaspth

### CODE REVIEW

File: guides/source/upgrading_ruby_on_rails.md [hunk 1]
```diff
@@ -272,6 +272,78 @@ You can invalidate the cache either by touching the product, or changing the cac
...
+    ]
+  end
+end
+```
+
+Instead of referring to `marginalia_controller`, any object can be stored in the `context` and then referenced in a tag definition. The default configuration will set the context with `controller` & `job` as appropriate.
+
+```ruby
+class MyController < ApplicationController
+  def controller_info
```
Comment (kaspth) [inline guides/source/upgrading_ruby_on_rails.md:309]: Can't have this as a public instance method, that'll expose it as an action. Can we point to other documentation instead that shows what's exposed in the context and when?

File: guides/source/upgrading_ruby_on_rails.md [hunk 2]
```diff
@@ -272,6 +272,78 @@ You can invalidate the cache either by touching the product, or changing the cac
 <% end %>
 ```
 
+#### Migrating from Marginalia
+
+Active Record ships with the same default configuration as Marginalia. For basic use, enable the feature in your application configuration:
+
+```ruby
+# config/application.rb
```
Comment (kaspth) [inline guides/source/upgrading_ruby_on_rails.md:280]: We typically don't want to overload the application configuration, so can we put the configurations in `config/initializers/query_log_tags.rb` instead? Later we can have new Rails 7 apps generate that by default.

Comment (keeran) [reply guides/source/upgrading_ruby_on_rails.md:280]: Sorry old habits - in general do we (app developers) not change the application config any more? 

Slightly thrown by this. If I wanted to enable the feature without any further customization of the tags, changing the config option in `config/application.rb` seems more appropriate than adding a new initializer, but I accept that I may be behind the times here! :)

File: guides/source/upgrading_ruby_on_rails.md [hunk 3]
```diff
@@ -272,6 +272,78 @@ You can invalidate the cache either by touching the product, or changing the cac
...
+```ruby
+class MyController < ApplicationController
+  def controller_info
+    "custom_information"
+  end
+end
+
+ActiveRecord::QueryLogs.update_context(server: "server_name")
+ActiveRecord::QueryLogs.tags = [
+  controller_info: -> { context[:controller].controller_info },
```
Comment (composerinteralia) [inline guides/source/upgrading_ruby_on_rails.md:316]: After https://github.com/rails/rails/pull/43079/files does this need to be:

```suggestion
  controller_info: ->(context) { context[:controller].controller_info },
```

File: guides/source/upgrading_ruby_on_rails.md [hunk 4]
```diff
@@ -272,6 +272,78 @@ You can invalidate the cache either by touching the product, or changing the cac
...
+  server: -> { context[:server] }
```
Comment (composerinteralia) [inline guides/source/upgrading_ruby_on_rails.md:317]: And:

```suggestion
  server: ->(context) { context[:server] }
```

**Comment by rafaelfranca** [comment]
@keeran do you want to finish this PR?

**Comment by keeran** [comment]
> do you want to finish this PR?

Hey - yes - tomorrow I will refresh on any changes which have been made since I started (thank you!), and update accordingly 👍

File: guides/source/upgrading_ruby_on_rails.md (hunk 2 above)
Comment (keeran) [reply guides/source/upgrading_ruby_on_rails.md:280]: Sorry to bring this up again @kaspth but I'm still thrown by where developers are expected to enable a feature. 

If someone wishes to enable default behaviour, would we still guide them towards creating an initializer?

If so, would they go for:

```
Rails.application.config.active_record.query_log_tags_enabled = true
```

**Pull request closed without merging**